test/e2e/issues_test.go

```diff
@@ -339,6 +339,57 @@ spec:
 		MustApplyYAML(eetestUpdated)
 		FieldShouldContainWithTimeout("eetests.e2e.hnc.x-k8s.io", nsChild, "eetest-sample", ".spec.foo", "bar", 30)
 	})
+
+	It("Should allow objects to be propagated only when a selector is set in AllowPropagate mode - issue #16", func() {
+		// Create a simple structure and get an object propagated
+		CreateNamespace(nsParent)
+		CreateNamespace(nsChild)
+		MustRun("kubectl hns set", nsChild, "--parent", nsParent)
+
+		MustRun("kubectl -n", nsParent, "create secret generic my-creds --from-literal=password=iamteamb")
+		// secret does not show up in child namespae because we haven’t configured HNC to propagate secrets in HNCConfiguration.
+		RunShouldNotContain("my-creds", defTimeout, "kubectl -n", nsChild, "get secrets")
+		// after setting AllowPropagate mode, the object does not get propogated
+		MustRun("kubectl hns config set-resource secrets --mode AllowPropagate --force")
+		time.Sleep(2 * time.Second)
+		RunShouldNotContain("my-creds", defTimeout, "kubectl -n", nsChild, "get secrets")
```

Comment:
An even better test might be to first change the mode to `Propagate`, verify that the secret is created, and then change it to `AllowPropagate` and verify that it's _removed_ before going on with the rest of the test.